save-frontend/src/main/kotlin/com/saveourtool/save/frontend/utils/RequestUtils.kt

```diff
@@ -143,25 +170,56 @@ suspend fun WithRequestStatusContext.delete(
  */
 @Suppress("MAGIC_NUMBER")
 internal fun Component<*, *>.classComponentResponseHandler(
-    response: Response
+    response: Response,
 ) {
-    // dirty hack to determine whether this component contains `setResponse` in its context.
-    // If we add another context with a function, this logic will break.
-    val hasResponseContext = this.asDynamic().context is Function<*>
+    val hasResponseContext = this.asDynamic().context is RequestStatusContext
     if (hasResponseContext) {
-        this.unsafeCast<Component<*, *>>().withModalResponseHandler(response)
+        this.withModalResponseHandler(response)
     }
 }
 
-private fun Component<*, *>.withModalResponseHandler(response: Response) {
+/**
+ * Handler that allows to show loading modal
+ *
+ * @param request REST API method
+ * @return [Response] received with [request]
+ */
+internal suspend fun Component<*, *>.loadingHandler(request: suspend () -> Response) = run {
+    val context: RequestStatusContext = this.asDynamic().context
+    context.setLoadingCounter { it + 1 }
+    val response = request()
+    context.setLoadingCounter { it - 1 }
+    response
+}
+
+/**
+ * Handler that allows to show loading modal
+ *
+ * @param request REST API method
+ * @return [Response] received with [request]
+ */
+@Suppress("MAGIC_NUMBER")
+internal suspend fun Component<*, *>.classLoadingHandler(request: suspend () -> Response): Response {
```

Comment:
Why do you have two functions for this? And if this one is intended as a wrapper to be used in components, then why is `Component<*, *>.loadingHandler` not private?